mkat_fpga_tests/test_cbf.py

```diff
@@ -583,4 +585,88 @@ def test_sensor_values(self):
         """
         (TP.C.1.16) Report sensor values (AR1)
         """
-        pass
+        iom = ioloop_manager.IOLoopManager()
+        iow = resource_client.IOLoopThreadWrapper(iom.get_ioloop())
+        iom.start()
+        self.addCleanup(iom.stop)
+
+        rc = resource_client.KATCPClientResource(
+            dict(name='localhost', address=('localhost', '7147'),
+                controlled=True))
+        rc.set_ioloop(iom.get_ioloop())
+        rct = resource_client.ThreadSafeKATCPClientResourceWrapper(rc, iow)
+        rct.start()
+        rct.until_synced()
+
+        ## 1. Request a list of available sensors using KATCP command
+        ## 2. Confirm the CBF replies with a number of sensor-list inform messages
+        LOGGER.info (rct.req.sensor_list())
+
+        # 3. Confirm the CBF replies with "!sensor-list ok numSensors"
+        #   where numSensors is the number of sensor-list informs sent.
+        list_reply, list_informs = rct.req.sensor_list()
+        sens_lst_stat, numSensors = list_reply.arguments
+        numSensors = int(numSensors)
+        self.assertEqual(numSensors, len(list_informs),
+            msg= ('Number of sensors are not equal to the'
+                'number of sensors in the list.'))
+
+        # 4. Test that ?sensor-value and ?sensor-list agree about the number
+        # of sensors.
+        sens_val_stat, sens_val_cnt = rct.req.sensor_value().reply.arguments
+        self.assertEqual(int(sens_val_cnt), numSensors,
+            msg='Sensors count are not the same')
+
+        # Sensors status
+        sens_status, sens_value = (rct.sensor.time_synchronised.get_reading()[2:4])
+        self.assertTrue(sens_value, msg='Sensor was not read successfully.')
+        # Sensors actual status
+        self.assertEqual(rct.sensor.time_synchronised.status,
+            'nominal', msg='Status Failed, current status:{}'
+                .format(rct.sensor.time_synchronised.status))
+
+        # 4. Request the time synchronisation status using KATCP command
+        #"?sensor-value time.synchronised
+        self.assertTrue(rct.req.sensor_value.issue_request(
+            'time.synchronised').reply.reply_ok(),
+                msg='Time Synchronisation Failed!')
+
+        # 5. Confirm the CBF replies with " #sensor-value <time>
+        # time.synchronised [status value], followed by a "!sensor-value ok 1"
+        # message.
+        self.assertEqual(str(
+            rct.req.sensor_value.issue_request('time.synchronised')[0]),
+                '!sensor-value ok 1', msg='Time Synchronisation Failed!')
+
+        # Check all sensors statuses
```

Comment:
Rather loop over the sensor objects (for sensor in rc.sensor.values(): ...) and the on each sensor object do a sensor.get_value() (which will poll the sensor), and then you can check sensor.status.